### PR TITLE
[Kimi-K3] Align forward numerics with official implementation

### PR DESCRIPTION
**Pull request by huangjiyi**
### PR Category
Distributed Strategy

### PR Types
Bug fixes

### Description
- 对齐 Kimi-K3 AttentionRes 的 FP32 batched reduction，并在 backward 后及时释放 PyLayer 保存的 activation。
- 对齐 KDA 的 causal SiLU conv、FP32 beta、in-kernel Q/K L2Norm 与 fused RMSNorm+sigmoid gate。
- 为 MLA 增加独立的 `qk_norm_eps` 配置，并保留对 `rms_norm_eps` 的默认回退。

### 是否引起精度变化
是

### CODE REVIEW

File: tests/single_card_tests/transformer/test_kimi_delta_attention.py [hunk 1]
```diff
@@ -405,6 +412,49 @@ def test_forward_always_calls_fla_chunk_kda(self):
...
+                kda_mod.fla.ops.kda,
+                "chunk_kda",
+                wraps=paddle_chunk_kda,
+            ),
+        ):
+            kda(hidden_states=x, attention_mask=None)
+
+        mock_silu.assert_called_once()
+
+    def test_fused_sigmoid_gated_norm_contract(self):
```
Comment (risemeup1111) [inline tests/single_card_tests/transformer/test_kimi_delta_attention.py:482]: P1

这里把新增的融合实现整体 mock 成任意张量，只验证了参数转发；即使 RMSNorm 公式、BF16/FP32 转换或 backward 梯度错误，测试仍会通过。现有全链路用例也只检查 shape/finite，无法证明本 PR 声明的数值对齐。

请增加不 mock `rms_norm_gated` 的前向和反向对齐测试：用显式 FP32 `rsqrt(mean(x^2) + eps) * weight * sigmoid(gate)` 作为独立 oracle，至少覆盖 FP32/BF16，并比较 `x`、`gate`、`weight` 的梯度。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
| 序号 | 位置 | 优先级 | 状态 |
|---|---|---|---|
| 1 | [KDA gated norm 兼容性](https://github.com/PaddlePaddle/PaddleFleet/pull/1651#discussion_r3728578732) | P1 | 🟡 |
| 2 | [KDA 融合数值测试](https://github.com/PaddlePaddle/PaddleFleet/pull/1651#discussion_r3728623569) | P1 | ✅ |

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: src/paddlefleet/transformer/kimi_delta_attention.py [hunk 2]
```diff
@@ -503,13 +501,15 @@ def forward(
...
-        x_dtype = x.dtype
-        x = x.reshape([-1, x.shape[-1]])
-        y = self.out_norm(x)
-        gate = gate.reshape([-1, gate.shape[-1]])
-        y = y * F.sigmoid(gate.astype(paddle.float32))
-        return y.astype(x_dtype)
+        """Official FLA fused per-head RMSNorm with a sigmoid output gate."""
+        return fla.modules.fused_norm_gate.rms_norm_gated(
+            x,
+            gate,
```
[the remark this answers is not in the document]
Comment (huangjiyi) [reply src/paddlefleet/transformer/kimi_delta_attention.py]: 和官方调用对齐，默认调用 fused 算子，性能更好，精度也能和官方对齐

File: tests/single_card_tests/transformer/test_kimi_delta_attention.py (hunk 1 above)
Comment (huangjiyi) [reply tests/single_card_tests/transformer/test_kimi_delta_attention.py:482]: 已按建议在 5c09a43 补充不 mock 的数值单测：FP32/BF16 均使用显式 FP32 RMSNorm + sigmoid oracle，并用同一 output gradient 比较 forward 以及 x、gate、weight 的梯度。完整 KDA 单测为 18 passed（新增用例含 2 个 dtype subtests），pre-commit、compileall 和 diff check 均通过。麻烦再帮忙看下。

Comment (risemeup1111) [reply tests/single_card_tests/transformer/test_kimi_delta_attention.py:482]: 已核对 5c09a43：新增测试确实移除了 rms_norm_gated mock，使用独立 FP32 RMSNorm+sigmoid oracle，覆盖 float32/BF16，并用同一 output gradient 比较 x、gate、weight 梯度；该测试覆盖 P1 已解决。当前仍有独立的兼容性问题：config.normalization 为 LayerNorm 时，KDA 的 out_norm 带 bias，但生产代码仍无条件走 fused RMSNorm 并忽略 bias；请一并修复该分支或明确限制配置后再合入。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
当前仍存在 Review Board 中的 KDA gated norm 兼容性 P1：`normalization="LayerNorm"` 时，构造出的带 bias norm 仍被无条件按 fused RMSNorm 执行并忽略 bias。请修复该分支或显式限制 KDA 仅支持 RMSNorm 后再合入。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
当前 head 仍未解决 KDA gated norm 兼容性 P1：`forward()` 在 `use_fused_kernels` 为真时无条件调用 `_apply_fused_gated_norm()`，而 `gpt_layer_specs.py` 在 `normalization="LayerNorm"` 时构造的是带 bias 的 LayerNorm；fused helper 仍将 bias 固定为 `None`，并且 `IdentityOp` 配置还可能没有 `weight`。请仅对 weight-only RMSNorm 走 fused 路径，其他 norm 回退 `_apply_gated_norm()`，或显式拒绝不支持的配置后再合入。

**Review by risemeup1111 — APPROVED** [review]

**Review by From00 — APPROVED** [review]
LGTM
